Add Zoom/pan to editor view.

markusng opened the pull request:
Add zooming, using scroll wheel and
middle mouse button to pan.

Manipulator class modified to support function.

Fix launching of dwpicker in Maya 2020 (<Python 3.6 version doesn't support f'' string formatting)

markusng commented:
Any notes for this? What can I do to make it mergeable?

luckylyk commented:
damn i missed this, let me try it out

luckylyk commented:
Ok, so firtly, i'm super sorry review this so lately, I've seen your post so many time, and though: "I already seen that".
But trying to clear my tickets, I realized that actually no, I never opened the ticket and only realise what it is about now :/

I tried your branch ! This a really nice work. Codewize I'm totally ok with the style.
Before to go further, if you are still motivated to collaborate the code, I encountered few bugs with your version. 

As you can see in the video: 
   - the selection square doesnt seems to work ?
   - The canvas does not resize with the scroll area then it cuts the edges
   - Middle click to navigate always place the canvas at strange coordinate when I start to manipulate.

https://github.com/user-attachments/assets/9cf592f4-ec7d-45a4-91ce-1ff5f3e033ca

If you are able to fix those steps, I would be more than happy to merde !

ahmettabak1 commented:
hi, pan zoom would be a really good improvement. will there be an update soon? @markusng @luckylyk

luckylyk commented:
@markusng Finally added the feature based on your code.
I fixed everything was going wrong and also removed the concept limiting the picker to a defined size (which was legacy from hotbox designer but not making sense for a picker)
Your code as conflict with new commits then i made some copy paste instead of merging it directly. But you are added in the "about" window as contributor :)

This is not committed / released yet but coming later tonight !
Thanks again for you contribution
Lio

markusng commented:
Thank you!  Much appreciated.  I was just starting to see I could fix the
odd center pivot bug.   Glad you've figured that out.

On Sun, Dec 8, 2024, 3:20 p.m. Lionel Brouyère ***@***.***>
wrote:

> @markusng <https://github.com/markusng> Finally added the feature based
> on your code.
> I fixed everything was going wrong and also removed the concept limiting
> the picker to a defined size (which was legacy from hotbox designer but not
> making sense for a picker)
> Your code as conflict with new commits then i made some copy paste instead
> of merging it directly. But you are added in the "about" window as
> contributor :)
>
> This is not committed / released yet but coming later tonight !
> Thanks again for you contribution
> Lio
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DreamWall-Animation/dwpicker/pull/97#issuecomment-2526362801>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADRDPZOEW45I4I73KEV5GD2ESSY5AVCNFSM6AAAAABKW46IIWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMRWGM3DEOBQGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
